libs/hwhandler.sh

```diff
@@ -83,23 +82,62 @@ get_libcamera_path() {
     fi
 }
 
+# print libcamera resolutions
+list_picam_resolution() {
+    local prefix
+    prefix="$(date +'[%D %T]') crowsnest:"
+    log_msg "'libcamera' device(s) resolution(s) :"
+    while read -r i; do
+        printf "%s\t\t%s\n" "${prefix}" "${i}" >> "${CROWSNEST_LOG_PATH}"
+    done < <(libcamera-hello --list-cameras | sed '1,2d;s/Modes:/Colorspace:/')
+}
+
+#type=3):
+get_libcamera_controls() {
+    local ust_bin flags
+    flags=( --camera-type=libcamera --camera-list_options )
+    ust_bin="${BASE_CN_PATH}/bin/camera-streamer/camera-streamer"
+    if [[ -x "${ust_bin}" ]]; then
+        "${ust_bin}" "${flags[@]}" 2> /dev/null | \
+        sed 's/device//g;/^SNAPSHOT/q' | sed '/^SNAPSHOT/d' | \
+        sed '/^CAMERA/d;/- property/d' | sed '/camera-streamer Version:/d' | \
+        sed 's/- available option: //g' | sed '/^$/d;' | \
+        sed 's/([0-9]*[a-z,0-9]\,//g' | sed '/type=7/d;/type=4/d' | \
+        sed 's/type=1/\t(bool/g;s/type=3/\t(int/g;s/type=5/\t(float/g' | \
+        sed 's/\[/\tmin=/g;s/\.\./\tmax=/g;s/\]$//g'
+    else
+        log_msg "WARN: 'libcamera' device option can not be displayed, because"
+        log_msg "WARN: camera-streamer is not installed"
+    fi
+}
+
+list_picam_controls() {
+    local prefix
+    prefix="$(date +'[%D %T]') crowsnest:"
+    log_msg "'libcamera' device controls :"
+    while read -r i; do
+        if [[ ! "${i}" =~ "INFO" ]]; then
+            printf "%s\t\t%s\n" "${prefix}" "${i}" >>"${CROWSNEST_LOG_PATH}"
+        fi
+    done < <(get_libcamera_controls)
+    # blank line workaround
+    log_msg ""
+}
+
 # Determine connected "legacy" device
 function detect_legacy {
     local avail
     if [[ "$(is_raspberry_pi)" = "1" ]] &&
-    command -v vcgencmd &> /dev/null; then
-        if vcgencmd get_camera &> /dev/null ; then
-            avail="$(vcgencmd get_camera \
-                    | awk -F '=' '{ print $3 }' \
-                    | cut -d',' -f1 \
-                    )"
+        command -v vcgencmd &>/dev/null; then
```

Comment:
`command -v vcgencmd &>/dev/null; then` shouldn't be intended as it belongs to the if.